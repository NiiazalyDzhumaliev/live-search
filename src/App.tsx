import { useState, useEffect } from 'react';
import styled from 'styled-components';
import axios from 'axios';
import DropDown from './DropDown';

const Container = styled.div``;
const Header = styled.h1``;
const InputContainer = styled.div``;
const Label = styled.label``;
const Input = styled.input``;

type Users = {
  id: number;
  name: string;
  username: string;
  email: string;
};

const App = () => {
  const [users, setUsers] = useState<Users[]>([]);
  const [loading, setLoading] = useState(true);
  const [searchName, setSearchName] = useState('');

  const handleInputChange = (e: { target: { value: string } }) => {
    const value = e.target.value;
    setSearchName(value);
  };

  const urlUsers = 'https://jsonplaceholder.typicode.com/users';
  useEffect(() => {
    const fetchResult = async () => {
      try {
        const result = await axios.get(urlUsers);
        setUsers(result.data);
        setLoading(false);
      } catch (error) {
        console.log(error);
      }
    };
    fetchResult();
  }, []);

  let filteredUsers = users.filter(user =>
    user.name.includes(searchName.toLowerCase())
  );

  filteredUsers.forEach(user => (user.name = user.name.toLowerCase()));

  return (
    <Container>
      {loading ? <Header>Loading...</Header> : <Header>LiveSearch</Header>}
      <InputContainer>
        <Label htmlFor="search-input">
          <Input
            type="text"
            id="search-input"
            placeholder="Search"
            onChange={e => handleInputChange(e)}
          />
        </Label>
      </InputContainer>
      <DropDown users={filteredUsers} />
    </Container>
  );
};

export default App;
