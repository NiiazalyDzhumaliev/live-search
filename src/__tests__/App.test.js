import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import App from '../App';

describe('App snapshot test', () => {
  test('testing app component', () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  test('renders correct content', () => {
    const root = document.createElement('div');
    ReactDOM.render(<App />, root);
    expect(root.querySelectorAll('[placeholder="Search"]')).not.toBeNull();
    expect(root.querySelector('input[type="text"]')).toBeTruthy();
  });
});
