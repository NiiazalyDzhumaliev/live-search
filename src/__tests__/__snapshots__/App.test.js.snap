// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test testing app component 1`] = `
<div
  className="sc-gKAaRy cOqsVP"
>
  <div
    className="sc-iCoGMd grrPaj"
  >
    <label
      className="sc-fujyAs"
      htmlFor="search-input"
    >
      <input
        className="sc-pNWdM bpZxol"
        id="search-input"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <img
        alt="magnify-glass"
        className="sc-jrsJWt IrIFW"
        src="loupe.svg"
      />
    </label>
  </div>
</div>
`;
